src/status_im2/contexts/communities/overview/view.cljs

```diff
@@ -12,164 +12,95 @@
-              :emoji "👽"}
-             {:name  "desktop-ui"
-              :emoji "👽"}
-             {:name  "desktop-ui-reviews"
-              :emoji "👽"}
-             {:name  "desktop2"
-              :emoji "👽"}
-             {:name  "desktop-ui2"
-              :emoji "👽"}
-             {:name  "desktop-ui2-reviews"
-              :emoji "👽"}]})
+(defn open-channel-token-gating-details
+  [name token-gating emoji channel-color]
+  (rf/dispatch
+   [:bottom-sheet/show-sheet
+    {:content
+     (fn []
+       [channel-token-gating-details name token-gating emoji channel-color])
+     :content-height 210}]))
 
-(defn channel-list-component-fn
-  [channel-heights first-channel-height]
-  [rn/view
-   {:on-layout #(swap! first-channel-height
-                  (fn []
-                    (+ (if platform/ios?
-                         0
-                         38)
-                       (int (Math/ceil (oops/oget % "nativeEvent.layout.y"))))))
-    :style     {:margin-top 20 :flex 1}}
-   (map-indexed (fn [index category]
-                  (let [first-category (first category)]
-                    ^{:key first-category}
-                    [rn/view
-                     {:flex      1
-                      :key       (str index first-category)
-                      :on-layout #(swap! channel-heights
-                                    (fn []
-                                      (sort-by :height
-                                               (conj @channel-heights
-                                                     {:height (int (oops/oget % "nativeEvent.layout.y"))
-                                                      :label  first-category}))))}
+(defn layout-y
+  [event]
+  (oops/oget event "nativeEvent.layout.y"))
 
-                     [quo/divider-label
-                      {:label            first-category
-                       :chevron-position :left}]
-                     [rn/view
-                      {:margin-left   8
-                       :margin-top    10
-                       :margin-bottom 8}
-                      (map-indexed (fn [inner-index channel-data]
-                                     [rn/view
-                                      {:key        (str inner-index (:name channel-data))
-                                       :margin-top 4}
-                                      [quo/channel-list-item channel-data]])
-                                   (first-category mock-list-of-channels))]]))
-                mock-list-of-channels)])
+(defn add-category-height
+  [categories-heights category height]
+  (swap! categories-heights
+    (fn []
+      (sort-by :height
+               (conj @categories-heights
+                     {:height height
+                      :label  category})))))
+
+(defn channel-list-component
+  [{:keys [on-categories-heights-changed
+           on-first-channel-height-changed]}
+   channels-list]
+  (let [categories-heights (reagent/atom [])]
```

Comment:
Shouldn't this local `categories-heights` state be defined using a form-2 component? Right now the state is recreated whenever `channel-list-component` renders. Even if this works right now, I think we should be extra cautious and redefine the component as a form-2 one.

Or of course maybe I got something wrong here.

Reply:
yes atoms should be outside of render function

Reply:
Here the behavior is intentional. We need `categories-heights` to be an atom, to store data after consecutive `on-layout` calls. But we need it to be recreated on rerender, because when`channels-list` property updated, all values of `categories-heights` should be discarded.

Reply:
Missed that, thanks @vkjr.

Upon further inspection, I can see this entire namespace is doing some rather inefficient stuff. We're doing a `swap!` on a reactive atom on every element of the `channels-list`, and then after every `swap!` another `swap!` after triggering `on-categories-heights-changed`. It looks like a lot of re-rendering work that doesn't scale well with the size of the list.

Unfortunately I don't any practical suggestion. You probably thought about this already.

Reply:
There are really a lot of questionable things in this namespace from the very beginning. My main suggestion is to fix them step by step, because otherwise, I will never close this PR :)))

Reply:
You're absolutely correct @vkjr. Baby steps! Or baby reps?

https://www.youtube.com/watch?v=Nnksg20hhvw